### PR TITLE
fix(docker): Safeguard config.json not being found

### PR DESCRIPTION
**Pull request by StealthyCoder**
This PR just adds a way to safeguard against the potential situation that someone just installs docker but never runs it yet. In that case there is no `.docker` dir created and no `config.json`.

In addition the reason the error checking is changed is because of an [internal comment of golang itself](https://pkg.go.dev/os#IsNotExist).

### CODE REVIEW

File: subcommands/common.go [hunk 1]
```diff
@@ -247,3 +247,10 @@ func FindWritableDirInPath(helperPath string) string {
 	}
 	return ""
 }
+
+func IsWritablePath(path string) string {
```
Comment (doanac) [inline subcommands/common.go:251]: An `Is*` function returning a string rather than a boolean is a little awkward. Looks like you are trying to save a couple of lines of code below, but I'd just make `isWritable` a public function and use that.

File: subcommands/docker/cmd.go [hunk 2]
```diff
@@ -86,9 +88,9 @@ func doDockerCreds(cmd *cobra.Command, args []string) {
 
 	var config map[string]interface{}
 	bytes, err := os.ReadFile(dockerConfigFile)
-	if os.IsNotExist(err) {
+	if errors.Is(err, fs.ErrNotExist) {
 		dockerConfig := filepath.Dir(dockerConfigFile)
-		if _, err := os.Stat(filepath.Dir(dockerConfig)); os.IsNotExist(err) {
+		if _, err := os.Stat(dockerConfig); errors.Is(err, fs.ErrNotExist) {
```
Comment (doanac) [inline subcommands/docker/cmd.go:93]: Is this really a 2nd change? i.e. the *fix* to your problem is checking if the path is writable. If so, `dockerConfigFile` becomes empty and a required flag. And this this is modernizing our error checking?

File: subcommands/common.go (hunk 1 above)
Comment (vkhoroz) [reply subcommands/common.go:251]: I concur on this. Other than this your PR LGTM.

File: subcommands/docker/cmd.go [hunk 3]
```diff
@@ -29,6 +29,9 @@ func NewCommand() *cobra.Command {
 	}
 	helperPath = subcommands.FindWritableDirInPath(helperPath)
 	dockerConfigFile = dockerConfigPath()
+	if !subcommands.IsWritable(dockerConfigFile) {
+		dockerConfigFile = ""
+	}
```
Comment (vkhoroz) [inline subcommands/docker/cmd.go:36]: So, there are 3 modernizr commits; and then this one line of code which _fixes the thing_, right?
Can you extract it into a separate commit with a loud message like _it fixes xyz_?

**Review by vkhoroz — APPROVED** [review]
LGTM

I think this is the case when _one merge commit_ to main is better than _3-4 commits created by rebase_.
Please, remember to hit the correct button when closing this PR (`Squash and merge` in GitHub terms).
